### PR TITLE
update numpy dtypes per guidance from numpy/pytest

### PR DESCRIPTION
**Pull request by ryancoe**
Updated dtypes per https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations. This is similar to https://github.com/LHEEA/meshmagick/pull/23.

All tests pass

```
$ pytest
========================================================================= test session starts ==========================================================================
platform darwin -- Python 3.9.2, pytest-6.2.3, py-1.10.0, pluggy-0.13.1
rootdir: /Users/rcoe/capytaine
plugins: hypothesis-6.9.1
collected 100 items                                                                                                                                                    

pytest/test_bem_engines.py ..                                                                                                                                    [  2%]
pytest/test_bem_green_functions.py ....                                                                                                                          [  6%]
pytest/test_bem_hierarchical_toeplitz_matrices.py ..........                                                                                                     [ 16%]
pytest/test_bem_linear_combination_of_dofs.py ..                                                                                                                 [ 18%]
pytest/test_bem_problems_and_results.py .........                                                                                                                [ 27%]
pytest/test_bem_solver.py ...                                                                                                                                    [ 30%]
pytest/test_bem_with_quadratures.py ..                                                                                                                           [ 32%]
pytest/test_bodies.py ..........                                                                                                                                 [ 42%]
pytest/test_bodies_predefined.py .....                                                                                                                           [ 47%]
pytest/test_consistency_with_Nemoh_2.py .......                                                                                                                  [ 54%]
pytest/test_io.py .                                                                                                                                              [ 55%]
pytest/test_io_meshes.py .                                                                                                                                       [ 56%]
pytest/test_matrices.py ...............                                                                                                                          [ 71%]
pytest/test_meshes.py ................                                                                                                                           [ 87%]
pytest/test_meshes_collections_and_symmetries.py ......                                                                                                          [ 93%]
pytest/test_meshes_geometry.py .....                                                                                                                             [ 98%]
pytest/test_post_pro_kochin.py .                                                                                                                                 [ 99%]
pytest/test_ui_vtk.py .                                                                                                                                          [100%]

========================================================================= 100 passed in 8.91s ==========================================================================
```

### CODE REVIEW

File: pytest/~test_io_pygmesh.py [hunk 1]
```diff
@@ -0,0 +1,564 @@
+import unittest
```
Comment (mancellin) [inline pytest/~test_io_pygmesh.py:1]: File committed by mistake?

File: pytest/test_ui_vtk.py [hunk 2]
```diff
@@ -11,6 +11,6 @@ def test_animation_of_dofs():
     try:
         animation = body.animate({"Heave": 0.2}, loop_duration=1.0)
         animation.embed_in_notebook()
-    except ImportError:
+    except ImportError as error:
```
Comment (mancellin) [inline pytest/test_ui_vtk.py:14]: Is it really needed?

**Comment by mancellin** [comment]
Thank you very much! Looks good to me except for the two comments above.

File: pytest/~test_io_pygmesh.py (hunk 1 above)
Comment (ryancoe) [reply pytest/~test_io_pygmesh.py]: yep ;)

5a18421

File: pytest/test_ui_vtk.py (hunk 2 above)
Comment (ryancoe) [reply pytest/test_ui_vtk.py]: no... not sure why I added this f7f41f8

**Pull request merged**